### PR TITLE
[dep] Use Rand type instead of deprecated rand.Seed

### PR DESCRIPTION
**Pull request by georgezgeorgez**
Determinstic random generator should not be set by rand.Seed
https://pkg.go.dev/math/rand#Seed
https://github.com/golang/go/issues/56319

### CODE REVIEW

**Comment by georgezgeorgez** [comment]
tests are still passing

**Review by MoonBaZZe — APPROVED** [review]

**Comment by 0x3639** [comment]
I will approve after @sumamu reviews.

**Comment by 0x3639** [comment]
@sumamu I think you missed this last PR.  Once you approve it so will I.  thx!

**Review by sumamu — APPROVED** [review]

**Comment by sumamu** [comment]
> @sumamu I think you missed this last PR. Once you approve it so will I. thx!

I looked into it a bit, then realized those were the unit tests so we're good to go.

**Review by 0x3639 — APPROVED** [review]

**Pull request merged**
